### PR TITLE
Change all "openspdm" to "libspdm"

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #46 
Change most instances of "openspdm" to "libspdm".
Make the the code meaning clearly.

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,19 +59,19 @@
 
 1) Presentation
 
-   Open Source Firmware Conference 2020 - [openspdm](https://cfp.osfc.io/osfc2020/talk/ECQ88N/)
```
Comment (steven-bellock) [inline README.md:62]: This can stay openspdm since that was the name at the time.

File: README.md [hunk 2]
```diff
@@ -59,19 +59,19 @@
 
 1) Presentation
 
-   Open Source Firmware Conference 2020 - [openspdm](https://cfp.osfc.io/osfc2020/talk/ECQ88N/)
+   Open Source Firmware Conference 2020 - [libspdm](https://cfp.osfc.io/osfc2020/talk/ECQ88N/)
 
-   Free and Open Source Developers European Meeting 2021 - [openspdm](https://fosdem.org/2021/schedule/event/firmware_uoifaaffsdc/)
```
Comment (steven-bellock) [inline README.md:64]: This can stay openspdm since that was the name at the time.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Wenxing-hou) [reply README.md:64]: Yes. I have changed. Thanks.

File: unit_test/test_spdm_responder/challenge_auth.c [hunk 3]
```diff
@@ -45,7 +45,7 @@ spdm_challenge_request_t    m_spdm_challenge_request6 = {
 };
 uintn m_spdm_challenge_request6_size = sizeof(m_spdm_challenge_request6);
 
-uint8_t m_opaque_challenge_auth_rsp[9] = "openspdm";
+uint8_t m_opaque_challenge_auth_rsp[9] = "libspdm";
```
Comment (jyao1) [inline unit_test/test_spdm_responder/challenge_auth.c:48]: this is wrong. please use sizeof.

File: unit_test/test_spdm_requester/challenge.c [hunk 4]
```diff
@@ -819,7 +819,7 @@ return_status spdm_requester_challenge_test_receive_message(
     // Ptr += spdm_get_hash_size (m_use_hash_algo);
     *(uint16_t *)Ptr = 8;
     Ptr += sizeof(uint16_t);
-    copy_mem (Ptr, "openspdm", 8);
+    copy_mem (Ptr, "libspdm", 8);
```
Comment (jyao1) [inline unit_test/test_spdm_requester/challenge.c:822]: this is wrong. please use sizeof.

Comment (Wenxing-hou) [reply unit_test/test_spdm_requester/challenge.c]: Thanks.  I have solved.

File: README.md (hunk 1 above)
Comment (Wenxing-hou) [reply README.md]: Yes. I have changed. Thanks.

**Comment by jyao1** [comment]
All feedback addressed. Merge.

**Pull request merged**